frb_codegen/src/config/opts.rs

```diff
@@ -29,17 +43,218 @@ pub struct Opts {
     pub bridge_in_method: bool,
     pub extra_headers: String,
     pub dart3: bool,
+    pub shared: bool, // it is true if this Opts instance is for auto-generated shared API block. Otherwise, it is false,
+    // for the below 2 fields:
+    // in single-block case, they should be `None`;
+    // in multi-blocks case, they should be paths to generated files with full directories
+    pub shared_rust_output_path: Option<String>,
+    pub shared_dart_output_path: Option<String>,
     pub keep_going: bool,
 }
 
 impl Opts {
-    pub fn get_ir_file(&self) -> ParserResult<IrFile> {
-        // info!("Phase: Parse source code to AST");
-        let source_rust_content = read_rust_file(&PathBuf::from(&self.rust_input_path));
-        let file_ast = syn::parse_file(&source_rust_content)?;
+    /// NOTE: for `Opts` for a regular API block, if `all_configs` is empty or with only 1 element,
+    /// it would return an `IrFile` with field of EMPTY `shared_types` for the regular block;
+    /// for `Opts` for an auto-generated shared API block, make sure `all_configs` has at least 2 items, each of which
+    /// is for the regular block. Otherwise, it would panic.
```

Comment:
nit: well it is a bit dynamic. is it possible we enforce this using types?

for example, say we already change `Opt` to `RegularOpt` and `SharedBlockOpt`, 

eg: (using https://stackoverflow.com/questions/32552593/is-it-possible-for-one-struct-to-extend-an-existing-struct-keeping-all-the-fiel)

```
struct BaseOpt { .. lots of fields .. }
struct RegularOpt {
  base: BaseOpt,
  .. maybe extra field, maybe not ..
}
struct SharedBlockOpt {
 .. similar ..
}
```